### PR TITLE
fix: incluster name resolve issue

### PR DESCRIPTION
**Pull request by fengxsong**
Signed-off-by: fengxsong <fengxsong@outlook.com>

When nginx listen with hostnetwork, change default `dnsPolicy` to `ClusterFirstWithHostNet`

### CODE REVIEW

File: charts/apisix/templates/_pod.tpl [hunk 1]
```diff
@@ -115,6 +115,10 @@ spec:
       {{- end }}
       resources:
       {{- toYaml .Values.apisix.resources | nindent 8 }}
+  {{- if .Values.apisix.hostNetwork }}
```
Comment (tokers) [inline charts/apisix/templates/_pod.tpl:118]: Also add this field to readme.

Comment (fengxsong) [reply charts/apisix/templates/_pod.tpl:118]: done, maybe we can use tools like https://github.com/norwoodj/helm-docs to generate README later?

File: charts/apisix/templates/_helpers.tpl [hunk 2]
```diff
@@ -96,17 +96,3 @@ prometheus:
 {{- define "apisix.pluginAttrs" -}}
 {{- merge .Values.pluginAttrs (include "apisix.basePluginAttrs" . | fromYaml) | toYaml -}}
 {{- end -}}
-
-{{- define "apisix.podAntiAffinity" -}}
```
Comment (tao12345666333) [inline charts/apisix/templates/_helpers.tpl:100]: Why delete this one?

File: charts/apisix/templates/_pod.tpl (hunk 1 above)
Comment (tao12345666333) [reply charts/apisix/templates/_pod.tpl:118]: Yes, I agree! Helm-docs is a good tool

File: charts/apisix/templates/_helpers.tpl (hunk 2 above)
Comment (fengxsong) [reply charts/apisix/templates/_helpers.tpl:100]: not need for this template function. as kube-scheduler will take care of this.

**Review by tokers — APPROVED** [review]

File: charts/apisix/templates/_helpers.tpl (hunk 2 above)
Comment (tao12345666333) [reply charts/apisix/templates/_helpers.tpl:100]: yes

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**